src/help/help_impl.cpp

```diff
@@ -838,27 +838,54 @@ void generate_races_sections(const config *help_cfg, section &sec, int level)
 		}
 	}
 
-	for(std::set<std::string, string_less>::iterator it = races.begin(); it != races.end(); ++it) {
+	// Support [race]help_taxonomy=, by adding all the races without that attribute
+	// first. This implementation assumes that there's only one level of nesting, but
+	// avoids a race condition in the order that races are added.
+	struct taxonomy_queue_type {
+		std::string id;
+		section content;
+	};
+	std::vector<taxonomy_queue_type> taxonomy_queue;
+
+	for(const auto& race_id : races) {
 		section race_section;
 		config section_cfg;
 
-		bool hidden = (visible_races.count(*it) == 0);
+		bool hidden = (visible_races.count(race_id) == 0);
 
-		section_cfg["id"] = hidden_symbol(hidden) + race_prefix + *it;
+		section_cfg["id"] = hidden_symbol(hidden) + race_prefix + race_id;
 
 		std::string title;
-		if (const unit_race *r = unit_types.find_race(*it)) {
+		std::string help_taxonomy;
+		if (const unit_race *r = unit_types.find_race(race_id)) {
 			title = r->plural_name();
+			help_taxonomy = r->help_taxonomy();
 		} else {
 			title = _ ("race^Miscellaneous");
+			// leave help_taxonomy empty
 		}
 		section_cfg["title"] = title;
 
-		section_cfg["sections_generator"] = "units:" + *it;
-		section_cfg["generator"] = "units:" + *it;
+		section_cfg["sections_generator"] = "units:" + race_id;
+		section_cfg["generator"] = "units:" + race_id;
 
 		parse_config_internal(help_cfg, &section_cfg, race_section, level+1);
-		sec.add_section(race_section);
+
+		if(help_taxonomy.empty()) {
+			sec.add_section(race_section);
+		} else {
+			taxonomy_queue_type x {hidden_symbol(hidden) + race_prefix + help_taxonomy, std::move(race_section)};
```

Comment:
That `hidden_symbol(hidden)` needs to compute `hidden` for the parent race, not the current one.

Reply:
It's now checking the parent's visibility, so this one's resolved; however, the question of whether the parent *should* be visible isn't resolved.